3_harmonize.R

```diff
@@ -1,25 +1,78 @@
 # Source the functions that will be used to build the targets in p3_targets_list
 source("3_harmonize/src/format_columns.R")
 source("3_harmonize/src/clean_wqp_data.R")
+source("3_harmonize/src/clean_conductivity_data.R")
+source("3_harmonize/src/clean_temperature_data.R")
 
 p3_targets_list <- list(
   
-  # All columns in p2_wqp_data_aoi are of class character. Coerce select columns
-  # back to numeric, but first retain original entries in new columns ending in 
-  # "_original". The default option is to format "ResultMeasureValue" and 
-  # "DetectionQuantitationLimitMeasure.MeasureValue" to numeric, but additional
-  # variables can be added using the `vars_to_numeric` argument in format_columns().
-  # By default, format_columns() will retain all columns, but undesired variables 
-  # can also be dropped from the WQP dataset using the optional `drop_vars` argument. 
+  # All columns in p2_wqp_data_aoi are of class character. Coerce select 
+  # columns back to numeric, but first retain original entries in new columns
+  # ending in "_original". The default option is to format "ResultMeasureValue"
+  # and "DetectionQuantitationLimitMeasure.MeasureValue" to numeric, but 
+  # additional variables can be added using the `vars_to_numeric` argument in 
+  # format_columns(). By default, format_columns() will retain all columns, but
+  # undesired variables can also be dropped from the WQP dataset using the 
+  # optional `drop_vars` argument. 
   tar_target(
     p3_wqp_data_aoi_formatted,
     format_columns(p2_wqp_data_aoi)
   ),
   
-  # Harmonize WQP data
+  # Harmonize WQP data by uniting diverse characteristic names under more
+  # commonly-used water quality parameter names, flagging missing records,
+  # and flagging duplicate records. Duplicated rows are identified using 
+  # the argument `duplicate_definition`. By default, a record will be 
+  # considered duplicated if it shares the same organization, site id, date,
+  # time, characteristic name and sample fraction, although a different 
+  # vector of column names can be passed to `clean_wqp_data()` below. By 
+  # default, duplicated rows are flagged and omitted from the dataset. To 
+  # retain duplicate rows, set the argument `remove_duplicated_rows` to FALSE. 
   tar_target(
     p3_wqp_data_aoi_clean,
     clean_wqp_data(p3_wqp_data_aoi_formatted, p1_char_names_crosswalk)
+  ),
+  
+  # Group the WQP data by parameter group in preparation for parameter-specific
+  # data cleaning steps.
+  tar_target(
+    p3_wqp_data_aoi_clean_grp,
+    p3_wqp_data_aoi_clean %>%
+      group_by(parameter) %>%
```

Comment:
```suggestion
      left_join(p3_wqp_param_cleaning_info, by = "parameter") %>% 
      group_by(parameter) %>%
```

Reply:
Great idea, it's so nice to make these builds more efficient ✨. I've included these changes in https://github.com/USGS-R/ds-pipelines-targets-example-wqp/pull/75/commits/8b387d9d44feaa9a3c498cfc59e0f47187f385d3.